RandomData.Api.Tests/WordGeneration/WordsGenerationTests.cs

```diff
@@ -0,0 +1,202 @@
+﻿using System;
+using FluentAssertions;
+using Microsoft.Extensions.Caching.Memory;
+using Newtonsoft.Json;
+using RandomData.Api.Exceptions;
+using RandomData.Api.Extensions.StringManipulation.Enums;
+using RandomData.Api.Services.Random.ServiceImplementations;
+using RandomData.Api.Tests.Services.FileReader;
+using RandomData.Api.WordGeneration.Configuration;
+using RandomData.Api.WordGeneration.Dto;
+using RandomData.Api.WordGeneration.Exceptions;
+using RandomData.Api.WordGeneration.Validators;
+using Xunit;
+
+namespace RandomData.Api.Tests.WordGeneration
+{
+    public class WordsGenerationTests
+    {
+        [Theory]
+        [InlineData(Format.Default, "Hello World")]
+        [InlineData(Format.Camel, "helloWorld")]
+        [InlineData(Format.Kebab, "hello-world")]
+        [InlineData(Format.Lower, "hello world")]
+        [InlineData(Format.Pascal, "HelloWorld")]
+        [InlineData(Format.Snake, "hello_world")]
+        [InlineData(Format.Upper, "HELLO WORLD")]
+        public void FormatProperty(Format format, string expectedOutput)
+        {
+            //arrange
+            var fakeFile = new FakeFileReader("[\"Hello World\"]");
+            var service = new Api.WordGeneration.WordGeneration(GetFakeOptions(), fakeFile, new RandomGenerator(), new GetWordParametersValidator(), new MemoryCache(new MemoryCacheOptions()));
+
+            //act
+            var result = service.GenerateRandomString(new GetWordParameters
+            {
+                Format = format
+            });
+
+            //assert
+            result.Should().Be(expectedOutput);
+        }
+
+        [Theory]
+        [InlineData(Encoding.Default, "Hello World")]
+        [InlineData(Encoding.Base64, "SGVsbG8gV29ybGQ=")]
+        public void EncodingProperty(Encoding encoding, string expectedOutput)
+        {
+            //arrange
+            var fakeFile = new FakeFileReader("[\"Hello World\"]");
+            var service = new Api.WordGeneration.WordGeneration(GetFakeOptions(), fakeFile, new RandomGenerator(), new GetWordParametersValidator(), new MemoryCache(new MemoryCacheOptions()));
+
+            //act
+            var result = service.GenerateRandomString(new GetWordParameters
+            {
+                Encoding = encoding
+            });
+
+            //assert
+            result.Should().Be(expectedOutput);
+        }
+
+        [Theory]
+        [InlineData((string) null)]
+        [InlineData("")]
+        [InlineData("{}")]
+        [InlineData("[Hamburger]")]
+        public void ConstructorShouldReturnInvalidWordsDictionaryExceptionOnInvalidJson(string content)
+        {
+            //arrange
+            var fakeFile = new FakeFileReader(content);
+            Action a = () => new Api.WordGeneration.WordGeneration(GetFakeOptions(), fakeFile, new RandomGenerator(), new GetWordParametersValidator(), new MemoryCache(new MemoryCacheOptions()));
```

Comment:
It's act part to arrange. a is not best name for variable Maybe CreateWordGeneration